lib/services/firebase_service.dart

```diff
@@ -247,14 +249,127 @@ class FirebaseService extends IFirebaseService {
     });
   }
+
+  Future<List<QueryDocumentSnapshot>> getNotificationList() async {
+    var data;
+    var storedUserId = UserStore.getValueFromStore('storedUserId');
+    var anSubcollection = await getLatestAnSubCollection(userId: storedUserId);
+    var patientState = anSubcollection["state"];
+    var checkState = enumToString(PatientState.postOperationHome);
+    if (patientState == checkState) {
+      data = await _firestore
+          .collection('Notifications')
+          .where('userId', isEqualTo: storedUserId)
+          .where('seen', isEqualTo: true)
+          .get();
+    } else {
+      data = await _firestore
+          .collection('Notifications')
+          .where('userId', isEqualTo: storedUserId)
+          .get();
+    }
+    return data.docs;
+  }
+
+  Future<List<Map<String, dynamic>>> getNotifications() async {
+    var notiList = await this.getNotificationList();
+    var returnList = notiList.map((user) async {
+      var notiCollection =
+          await _firestore.collection("Notifications").doc(user.id).get();
+      var seen = notiCollection['seen'];
+      if (seen == false) {
+        seen = "ยังไม่ได้ดำเนินการ";
+      } else {
+        seen = "ดำเนินการแล้ว";
+      }
+      var patientState = notiCollection['patientState'];
+      var formName = notiCollection['formName'];
+      formName = formNameModel[formName];
+      var time = notiCollection['creation'];
+      var formTime =
+          DateTime.fromMicrosecondsSinceEpoch(time.microsecondsSinceEpoch);
+      var formDateToShow = DateFormat('dd/MM/yyyy').format(formTime);
+      var formTimeToShow = "${DateFormat.Hm().format(formTime)} น.";
+
+      var patientSeen = notiCollection['patientSeen'] ?? '-';
+      if (patientSeen == false) {
+        patientSeen = "ยังไม่ได้อ่าน";
+      } else {
+        patientSeen = "อ่านแล้ว";
+      }
+      var map = {
+        'patientState': patientState ?? '-',
+        'formName': formName ?? '-',
+        'formTime': formTimeToShow ?? '-',
+        'formDate': formDateToShow ?? '-',
+        'formDateTimeSort': formTime ?? '-',
+        'seen': seen ?? '-',
+        'imgURL': '-',
+        'advice': '-',
+        'severity': '-',
+        'notiId': user.id ?? '-',
+        'patientSeen': patientSeen ?? '-',
+      };
+      if (patientState == "Post-Operation@Home") {
+        var imgURL = notiCollection['imgURL'] ?? '-';
+        var advice = notiCollection['advice'] ?? '-';
+        var severity = notiCollection['severity'] ?? '-';
+        map.addAll({
+          'imgURL': imgURL ?? '-',
+          'advice': advice ?? '-',
+          'severity': severity ?? '-',
+        });
+      }
+      return map;
+    });
+    var futureList = Future.wait(returnList);
+    var returnValue = await futureList;
+    if (returnValue != null) {
+      returnValue.removeWhere((element) => element == null);
+    }
+    return returnValue;
+  }
+
+  Future<int> getNoticounter() async {
```

Comment:
rename to `getNotiCounter` . Remind to always use camelCase

Reply:
Solved ka